controllers/eventingauth_controller_test.go

```diff
@@ -35,97 +35,92 @@ var _ = Describe("EventingAuth Controller happy tests", Serial, Ordered, func()
 		}
 	})
 
-	AfterEach(func() {
-		cleanup()
+	AfterAll(func() {
+		revertIasNewClientStub()
 	})
 
-	Context("Creating EventingAuth CR", func() {
-
-		It("should create secret with IAS applications credentials", func() {
-			cr := createEventingAuthAndKubeconfigSecret()
-			verifyEventingAuthStatusReady(cr)
-			verifySecretExistsOnTargetCluster()
-		})
+	BeforeEach(func() {
+		crName = generateCrName()
+		createKubeconfigSecret(crName)
 	})
 
-	Context("Deleting EventingAuth CR", func() {
-
-		It("should delete secret with IAS applications credentials", func() {
-			cr := createEventingAuthAndKubeconfigSecret()
-			verifyEventingAuthStatusReady(cr)
-			verifySecretExistsOnTargetCluster()
-			deleteEventingAuth(cr)
-			verifySecretDoesNotExistOnTargetCluster()
-		})
+	AfterEach(func() {
+		deleteEventingAuth(eventingAuth)
+		deleteKubeconfigSecret(crName)
 	})
 
-	Context("Time-based EventingAuth CR reconciliation", func() {
+	Context("Creating and Deleting EventingAuth CR", func() {
```

Comment:
Why combined the tests for creation and deletion? Why not test each case separately?

Reply:
the reason was it is anyway deleting in both cases as each must clean-up the resources. I thought I just put both in one case instead of duplicating execution with different test case titles.